src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -70,6 +70,150 @@ public static NuGetFramework Parse(string folderName, IFrameworkNameProvider map
             return framework;
         }
 
+        /// <summary>
+        /// Creates a NuGetFramework from individual components
+        /// </summary>
+        public static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion)
+        {
+            return ParseComponents(targetFrameworkIdentifier, targetFrameworkVersion, targetFrameworkProfile, targetPlatformIdentifier, targetPlatformVersion, DefaultFrameworkNameProvider.Instance);
+        }
+
+        /// <summary>
+        /// Creates a NuGetFramework from individual components, using the given mappings
+        /// </summary>
+        internal static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion, IFrameworkNameProvider mappings)
+        {
+            if (string.IsNullOrEmpty(targetFrameworkIdentifier))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, "targetFrameworkIdentifier");
```

Comment:
Use `ArgumentNullException`, and we don't have to create our own message string.  (plus `nameof(targetFrameworkIdentifier)`)

Reply:
This was done at @dtivel's behest: https://github.com/NuGet/NuGet.Client/pull/3549#discussion_r463336057

also, we can't use `nameof` in `NuGet.Frameworks` because that needs C# 6.